### ISSUE TITLE
[Core] Provision an A10 GPU on Azure takes 20 minutes

### ISSUE THREAD

**Issue by cblmemo** [skypilot-org/skypilot#3718]
After #3707, an A10 cluster on Azure will take more than 20 minutes to provision, mainly due to a driver installation which takes ~12 minutes. We should investigate how to optimize this.

Provision time on #3707:

||Without Driver Installation|With Driver Installation|
|-|-|-|
|8 vCPU|4m52s|5m20s|
|A10|5m37s|21m15s|

**Comment by WesleyYue** [comment]
FWIW, installing the drivers manually gets me 10m8.160s

```bash
#!/bin/bash

# Update and install build-essential
sudo apt update
sudo apt install -y build-essential

nvidia-smi
echo "Installing GRID driver..."

# Download and install GRID driver
GRID_DRIVER_URL="https://download.microsoft.com/download/8/d/a/8da4fb8e-3a9b-4e6a-bc9a-72ff64d7a13c/NVIDIA-Linux-x86_64-535.161.08-grid-azure.run"
GRID_DRIVER_FILE="NVIDIA-Linux-x86_64-535.161.08-grid-azure.run"

wget -nv $GRID_DRIVER_URL -O $GRID_DRIVER_FILE
sudo chmod +x $GRID_DRIVER_FILE
sudo sh $GRID_DRIVER_FILE --silent

nvidia-smi
echo "Installing CUDA toolkit..."

# Download and install CUDA toolkit
CUDA_TOOLKIT_URL="https://developer.download.nvidia.com/compute/cuda/12.2.0/local_installers/cuda_12.2.0_535.54.03_linux.run"
CUDA_TOOLKIT_FILE="cuda_12.2.0_535.54.03_linux.run"

wget -nv $CUDA_TOOLKIT_URL -O $CUDA_TOOLKIT_FILE
sudo sh $CUDA_TOOLKIT_FILE --silent --toolkit --override

# Set environment variables
echo 'export PATH=$PATH:/usr/local/cuda-12.2/bin' >> ~/.bashrc
echo 'export LD_LIBRARY_PATH=/usr/local/cuda-12.2/lib64' >> ~/.bashrc

# Source the updated bashrc to apply changes
source ~/.bashrc

# Verify installations
nvidia-smi
nvcc -V

echo "Installation complete. Please verify that the GRID driver and CUDA toolkit have been installed successfully."

```

**Comment by Michaelvll** [comment]
This has been fixed by #4205
